common/fs/src/cache/dir_path.rs

```diff
@@ -74,3 +75,59 @@ impl From<DirPathBuf> for PathBuf {
         d.inner
     }
 }
+
+// Recursive directory back off
+fn find_valid_path(
+    path: Option<PathBuf>,
+    postfix: Option<PathBuf>,
+) -> Result<DirPathBuf, DirPathBufError> {
+    match path {
+        Some(p) => {
+            if p.is_dir() {
```

Comment:
does it work for:
- it is a file
- permission error

Reply:
The configuration is meant to be set as directory so a single file wouldn't be expected. I followed same pattern as existing directory check. As for permissions, in theory, I think could fail is for some reason it was running in non-root mode and didn't have directory access.